webconsole/src/main/java/org/secondbase/webconsole/HttpWebConsole.java

```diff
@@ -0,0 +1,135 @@
+
+    private final ServiceLoader<Widget> widgets
+            = ServiceLoader.load(Widget.class);
+
+    /**
+     * Basic /health endpoint, returning 200 OK.
+     */
+    private class HealthzHandler implements HttpHandler {
+        public void handle(final HttpExchange t) throws IOException {
+            final byte [] response = "Healthy".getBytes();
+            t.sendResponseHeaders(200, response.length);
+            final OutputStream os = t.getResponseBody();
+            os.write(response);
+            os.close();
+        }
+    }
+
+    /**
+     * Set up the webconsole using port from {@link WebConsoleConfiguration}.
+     * @throws IOException if server can't start on a given port.
+     */
+    public HttpWebConsole() throws IOException {
+        server = HttpServer.create();
+        server.createContext("/healthz", new HealthzHandler());
+    }
+
+    /**
+     * Load WebConsole Flags and set the secondbase webconsole to "this".
+     * @param secondBase module coordinator.
+     */
+    @Override
+    public void load(final SecondBase secondBase) {
+        secondBase.getFlags().loadOpts(WebConsoleConfiguration.class);
+        secondBase.setWebConsole(this);
+    }
+
+    @Override
+    public void start() throws IOException {
+        if (! WebConsoleConfiguration.enableWebConsole) {
+            return;
+        }
+        port = (WebConsoleConfiguration.port == 0) ? getFreePort() : WebConsoleConfiguration.port;
+        final int USE_SYSTEM_DEFAULT_BACKLOG = 0;
+        server.bind(
+                new InetSocketAddress(port),
+                USE_SYSTEM_DEFAULT_BACKLOG);
+        LOG.info("Starting webconsole on port " + port);
+        for (final Widget widget : widgets) {
+            LOG.info("Adding webconsole widget " + widget.getPath());
+            server.createContext(widget.getPath(), widget.getServlet());
+        }
+        server.start();
+    }
+
+    @Override
+    public void shutdown() throws IOException {
+        LOG.info("Shutting down webconsole.");
+        if (! WebConsoleConfiguration.enableWebConsole) {
+            return;
+        }
+        final int USE_SYSTEM_DEFAULT_BACKLOG = 0;
+        server.bind(
+                new InetSocketAddress(port),
+                USE_SYSTEM_DEFAULT_BACKLOG);
+        server.start();
+    }
+
+    @Override
+    public int getPort() {
+        return port;
+    }
+
+    /**
+     * Get the server implementation.
+     * @return HttpServer
+     */
+    public HttpServer getServer() {
+        return server;
+    }
+
+    /**
+     * Find a free local port. Strictly speaking I suspect this
+     * method of doing things has all sorts of race conditions but
+     * since ports allocated this way have a tendency to increment on
+     * the platforms I have tested it on it should for the most part
+     * work.
+     *
+     * @return a free local port number.
+     * @throws RuntimeException on error allocating port.
+     */
+    private int getFreePort() {
```

Comment:
I'm not sure about this. It doesn't sound like we're convinced about it. Maybe we can get away with not supporting auto-port search? At least reduce the amount of "personal touch" in the comments?